infra/modules/providers/azure/app-service/main.tf

```diff
@@ -14,25 +17,34 @@ data "azurerm_app_service_plan" "appsvc" {
 }
 
 resource "azurerm_app_service" "appsvc" {
-  name                = "${lower(element(keys(var.app_service_name), count.index))}-${lower(terraform.workspace)}"
+  name                = "${lower(element(keys(var.app_service_config), count.index))}-${lower(terraform.workspace)}"
```

Comment:
if taking the above suggestion this would look like:
`format("%s-%s", lower(locals.app_names[count.index]), lower(terraform.workspace))`


Also the use of `element()` is unnecessary in terraform 12 I think. This could become:
`"${lower(keys(var.app_service_config)[count.index])}-${lower(terraform.workspace)}"`